sto/cli/main.py

```diff
@@ -605,6 +611,122 @@ def kyc_manage(config: BoardCommmadConfiguration, whitelist_address):
+def voting_deploy(
+        config: BoardCommmadConfiguration,
+        token_address,
+        kyc_address,
+        voting_name,
+        uri,
+        type,
+        options
+):
+    """
+    Deploys Voting contract to desired ethereum network
+    network, ethereum-abi-file, ethereum-private-key, ethereum-node-url are required args
+    """
+    from sto.ethereum.utils import deploy_contract, integer_hash, get_contract_deployed_tx
+    from eth_utils import to_bytes
+    if kyc_address is None:
+        tx = get_contract_deployed_tx(config.dbsession, 'BasicKYC')
+        if not tx:
+            raise Exception('BasicKYC contract not deployed. Please call ')
+        kyc_address = tx.contract_address
+    args = {
+        '_token': token_address,
+        '_KYC': kyc_address,
+        'name': to_bytes(text=voting_name),
+        'URI': to_bytes(text=uri),
+        '_type': type,
+        '_hash': integer_hash(type),
+        '_options': [to_bytes(i) for i in options]
+    }
+    deploy_contract(config, contract_name='VotingContract', constructor_args=args)
+
+
+@cli.command(name="payout-deploy")
+@click.option('--token-address', required=True, help="address of security token contract", type=str)
+@click.option('--payout-token-address', required=True, help="address of payout token contract", type=str)
+@click.option('--kyc-address', required=False, default=None, help="address of kyc contract", type=str)
+@click.option('--payout-name', required=True, help="name of the payout,", type=str)
+@click.option('--uri', required=True, help="announcement uri", type=str)
+@click.option('--type', required=True, help="announcement type", type=int)
+@click.option('--options', required=False, default=[], help="additional payout contract options", type=list)
+@click.pass_obj
+def payout_deploy(
+        config: BoardCommmadConfiguration,
+        token_address,
+        payout_token_address,
+        kyc_address,
+        payout_name,
+        uri,
+        type,
+        options
+):
+    """
+    Deploys Voting contract to desired ethereum network
+    network, ethereum-abi-file, ethereum-private-key, ethereum-node-url are required args
+    """
+    from sto.ethereum.utils import deploy_contract, integer_hash, get_contract_deployed_tx
+    from eth_utils import to_bytes
+    if kyc_address is None:
+        tx = get_contract_deployed_tx(config.dbsession, 'BasicKYC')
+        if not tx:
+            raise Exception('BasicKYC contract not deployed. Please call ')
+        kyc_address = tx.contract_address
+    args = {
+        '_token': token_address,
+        '_payoutToken': payout_token_address,
+        '_KYC': kyc_address,
+        'name': to_bytes(text=payout_name),
+        'URI': to_bytes(text=uri),
+        '_type': type,
+        '_hash': integer_hash(type),
+        '_options': [to_bytes(i) for i in options]
+    }
+    deploy_contract(config, contract_name='PayoutContract', constructor_args=args)
+
+
+@cli.command(name="payout-deposit")
```

Comment:
Originally I was thinking that this command could take care of the approve() as well, but I guess it's cleaner this way :slightly_smiling_face: